server/src/com/thoughtworks/go/server/controller/JobController.java

```diff
@@ -192,6 +196,15 @@ public ModelAndView handleRequest(@RequestParam(value = "pipelineName") String p
         return jsonFound(json).respond(response);
     }
 
+    private AgentConfig agentConfig(String agentUuid) {
+        if (isBlank(agentUuid)) {
+            return null;
+        }
+
+        final AgentInstance agentInstance = agentService.findAgent(agentUuid);
+        return agentInstance == null ? null : agentInstance.agentConfig();
```

Comment:
That null check might not work as `agentService.findAgent(agentUuid)` returns a `NullAgentInstance` object if it cannot find a matching agent, see [this]( https://github.com/bdpiparva/gocd/blob/f31eeae5f61ce974be4d679f1b7d6b1f4c5ed63c/common/src/com/thoughtworks/go/server/domain/AgentInstances.java#L90). Might want to change that to `agentInstance.isNullAgent()`